### ISSUE TITLE
Proxmox VM, W11 24H2, Virtio Drivers - Install Hangs

### ISSUE THREAD

**Issue by TechBrain64** [cschneegans/unattend-generator#119]
I'm testing the auto-unattended deployment of Windows 11 24H2 using the Microsoft evaluation ISO on Proxmox 8.3.0. I followed the ["To install Windows on a virtual machine in Proxmox VE"](https://schneegans.de/windows/unattend-generator/usage/). The initial boot process kickoff without a hitch and immediately began to install. After a reboot, input computername, continued to script executions where after several minutes waiting, the Virtio-win 0.1.262 driver hung.

 
![Virtio-win-driver_Hangs](https://github.com/user-attachments/assets/9247d8df-0c5e-468d-84f6-d7cedaf9366d)

The "VirtioGuestTools.log" file was empty.

I cancelled out of the error, install completed, was able to install tools afterwards.

**Comment by jclendineng** [comment]
I can confirm this started happening in 24H2, I tested many different configs, I also tried moving the script to the first run but same result, it hangs. Worked great with previous version of windows, I tested running the install manually in powershell and it ran as expected, I changed the script to install just the MSI virtio in the autounattend.xml and that also worked (that installs drivers only, not guest tools) so its specifically something it doesn't like in the EXE.

**Comment by TechBrain64** [comment]
Can the two be installed separately?

**Comment by jclendineng** [comment]
I'm creating a 26100.2448 build now to test - Looks like 26100.2454 had 2 versions released due to a Microsoft bug.

**Comment by cschneegans** [comment]
Thanks for your report! I don't have a Proxmox environment available right now, so I am dependent on reports like yours to identify such problems.

There has been a very similar issue with VMware Tools, which I have [addressed](https://github.com/cschneegans/unattend-generator/commit/e5d634b81068267e477d3fb2d1a8b4fa315cfa50) recently: On Windows 11 24H2, VMware Tools installation hangs when done in the _Specialize_ phase, but succeeds when done in the _FirstLogon_ phase. It also succeeds in the _Specialize_ phase if Windows Defender is disabled.

Therefore, you could try one of the following:

- Select the _Disable Windows Defender_ checkbox

-or-

- Unselect the _Install VirtIO Guest Tools and QEMU Guest Agent_ checkbox
- Add the content of [VirtIoGuestTools.ps1](https://github.com/cschneegans/unattend-generator/blob/master/resource/VirtIoGuestTools.ps1) to the _Scripts to run when the first user logs on after Windows has been installed_ section as a .ps1 file. This will effectively move the VirtIO Guest Tools installation from the _Specialize_ to the _FirstLogon_ phase.

**Comment by TechBrain64** [comment]
I will give your suggestions a shot and follow-up with you later today.

**Comment by TechBrain64** [comment]
MS Defender was the culprit. As you've suggested, it appears to be impeding the VirtioGuestTools installation. Which leaves to questions: how to rectify and get Defender re-enabled.

I was wondering, if possible and feasible, would adding whatever files/folders exclusions for Virtio to Defender can overcome the hang during deployment?

If not, how do we re-enable Defender?

**Comment by jclendineng** [comment]
That works.

Add this to the scripts to run at login (ps1)

`& {
	foreach ( $letter in 'DEFGHIJKLMNOPQRSTUVWXYZ'.ToCharArray() ) {
		$exe = "${letter}:\virtio-win-guest-tools.exe";
		if ( Test-Path -LiteralPath $exe ) {
			Start-Process -FilePath $exe -ArgumentList '/passive', '/norestart' -Wait;
			return;
		}
	}
	'VirtIO Guest Tools image (virtio-win-*.iso) is not attached to this VM.';
} *>&1 >> 'C:\Windows\Setup\Scripts\VirtIoGuestTools.log';`

**Comment by cschneegans** [comment]
So it seems that VMware Tools and VirtIO Guest Tools are both blocked by Windows Defender in the _Specialize_ phase. 
https://github.com/cschneegans/unattend-generator/commit/fa900ac177ca2c724ef7d5c8bf9d4ec6e03a61c8 applies the same strategy to VirtIO Guest Tools installation that existed for VMware Tools before. Please refresh your autounattend.xml and repeat your Windows installation.

Note to self: Also test VirtualBox Guest Additions.

**Comment by TechBrain64** [comment]
Good news: I believe the issue has been resolved. I was able to deploy using your changes without issue, twice. In fact, the VirtioGuestTools now install with Defender enabled and silently.

PS: I found another issue, but I'll do separate write-up.

**Comment by cschneegans** [comment]
> VirtualBox Guest Additions 7.1.4r165100 was successfully installed in the _Specialize_ phase on Windows 11 24H2, so no need to move this script to the _FirstLogon_ phase.

I have now repeatedly experienced that the installation of the current release of VirtualBox Guest Additions (7.1.8) does indeed hang when Windows Defender is running. Therefore, will also move its installation to the _FirstLogon_ phase.
